File: typescript/base-hmi-types/src/screens/base/HmiColorAnimationState.ts
import { HmiColor } from "./HmiColor.js";
import { HmiColorBehavior } from "./HmiColorBehavior.js";
import { HmiGradientDirection } from "./HmiGradientDirection.js";

export class HmiColorAnimationState {
  value?: number;
  foregroundBehavior?: HmiColorBehavior;
  foregroundColor1?: HmiColor;
  foregroundColor2?: HmiColor;
  backgroundBehavior?: HmiColorBehavior;
  backgroundColor1?: HmiColor;
  backgroundColor2?: HmiColor;
  fillColorMode?: number;
  backgroundEndColor?: HmiColor;
  backgroundGradientStop?: number;
  backgroundGradientAxis?: string;
  backgroundGradientDirection?: HmiGradientDirection;
  fillEndColor?: HmiColor;
  fillGradientStop?: number;
  fillGradientAxis?: string;
  fillGradientDirection?: HmiGradientDirection;
}

File: src/BaseHmiTypes/Screens/Base/HmiColorAnimationState.cs
namespace BaseHmiTypes.Screens.Base;

public class HmiColorAnimationState
{
    public double? Value { get; set; }

    public HmiColorBehavior? ForegroundBehavior { get; set; }

    public HmiColor? ForegroundColor1 { get; set; }

    public HmiColor? ForegroundColor2 { get; set; }

    public HmiColorBehavior? BackgroundBehavior { get; set; }

    public HmiColor? BackgroundColor1 { get; set; }

    public HmiColor? BackgroundColor2 { get; set; }

    public int? FillColorMode { get; set; }

    public HmiColor? BackgroundEndColor { get; set; }

    public double? BackgroundGradientStop { get; set; }

    public string? BackgroundGradientAxis { get; set; }

    public HmiGradientDirection? BackgroundGradientDirection { get; set; }

    public HmiColor? FillEndColor { get; set; }

    public double? FillGradientStop { get; set; }

    public string? FillGradientAxis { get; set; }

    public HmiGradientDirection? FillGradientDirection { get; set; }
}
